packages/plugin-hardhat/src/utils/options.ts

```diff
@@ -32,16 +32,29 @@ type Initializer = {
   initializer?: string | false;
 };
 
-export type DeployBeaconProxyOptions = ProxyKindOption & Initializer;
-export type DeployBeaconOptions = StandaloneOptions;
-export type DeployImplementationOptions = StandaloneOptions & GetTxResponse;
-export type DeployProxyAdminOptions = DeployOpts;
-export type DeployProxyOptions = StandaloneOptions & Initializer;
+export type Platform = {
+  platform?: boolean;
+};
+
+export type PlatformSupportedOptions = Platform & {
+  verifySourceCode?: boolean;
+};
+
+export type DeployBeaconProxyOptions = DeployOpts & ProxyKindOption & Initializer & PlatformSupportedOptions;
```

Comment:
What's the reason `DeployOpts` was added here but not in all `Deploy*Options` types?

Reply:
The other `Deploy*Options` types inherit `StandaloneOptions` which already has `DeployOpts`.  Added doc comments to clarify some of the inherited options.